src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -616,10 +617,17 @@ private static class Watcher implements Runnable {
         @Override public void run() {
             try {
                 Integer exitStatus = controller.exitStatus(workspace, listener); // check before collecting output, in case the process is just now finishing
-                long lastLocation = 0;
-                FilePath lastLocationFile = controller.getLastLocationFile(workspace);
-                if (lastLocationFile.exists()) {
-                    lastLocation = Long.parseLong(lastLocationFile.readToString());
+                if (lastLocation == -1) {
+                    FilePath lastLocationFile = controller.getLastLocationFile(workspace);
+                    if (lastLocationFile.exists()) {
+                        lastLocation = Long.parseLong(lastLocationFile.readToString());
+                        LOGGER.finest(() -> "Loaded lastLocation=" + lastLocation);
```

Comment:
Seems to actually be missing from test coverage in this plugin currently; `workflow-durable-task-step` does exercise it in tests relating to agent disconnections and controller restarts.